File: basic/src/main/java/com/yuanshenbin/basic/state/EmptyState.java
package com.yuanshenbin.basic.state;


import android.support.annotation.DrawableRes;

/**
 * author : yuanshenbin
 * time   : 2017/12/7
 * desc   : 空数据状态
 */

public abstract class EmptyState extends StateAbstract {
    protected OnRetryListener listener;

    public abstract void showEmpty(@DrawableRes int resId, String... msg);

    public abstract void showEmpty(@DrawableRes int resId);

    public abstract void showEmpty(CharSequence... msg);
}

File: basic-kotlin/src/main/java/com/yuanshenbin/basic/state/EmptyState.kt
package com.yuanshenbin.basic.state

import androidx.annotation.DrawableRes

/**
 * author : yuanshenbin
 * time   : 2017/12/7
 * desc   : 空数据状态
 */
abstract class EmptyState : StateAbstract() {
    protected var listener: OnRetryListener? = null
    abstract fun showEmpty(@DrawableRes resId: Int, vararg msg: String?)
    abstract fun showEmpty(@DrawableRes resId: Int)
    abstract fun showEmpty(vararg msg: CharSequence?)
}